datalad/support/annexrepo.py

```diff
@@ -1120,6 +1156,20 @@ def _call_annex_records(self, args, files=None, jobs=None,
             #        args, shorten(e.stderr)
             #    )
 
+        # git-annex fails to non-zero exit when reporting an error on
+        # non-existing paths in some versions. Hence, check it on
+        # non-failure, too.
+        # TODO: Limit to versions that require this. (Was
+        # fixed according to https://github.com/datalad/datalad/pull/6510#issuecomment-1054499339)
+        if external_versions['cmd:annex'] >= '10.20220128':
```

Comment:
note: we also have `_check_version_kludges` where we at some point started to collect "named" kludges/changes in git-annex to react to.  I wonder if it would make sense to keep going with it and centralize our "knowledge" of git-annex behavior changes in there?

Reply:
kludges are note needed with current approach. The only conditionals of that kind are in the tests now.